cns/azure-cns.yaml

```diff
@@ -16,47 +68,96 @@ spec:
       annotations:
         cluster-autoscaler.kubernetes.io/daemonset-pod: "true"
     spec:
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: kubernetes.azure.com/cluster
```

Comment:
I think we should remove this key, I'm not sure we want to require a particular annotation that is not well-known, e.g. `node-role.kubernetes.io/master`)

Reply:
This was there already from Wei's change awhile back I'm not sure why though
https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp/pullrequest/2928590?_a=files&path=%2Fccp%2Fcharts%2Fkube-control-plane%2Fcharts%2Fkube-addons%2Ftemplates%2Faddons%2F_azure-cns.yaml

Reply:
Checked with Wei and this label is from aks-engine so it's present on all nodes
https://github.com/Azure/aks-engine/blob/0ca3854c98cc9d1386a0339949c155fe636a6184/pkg/api/types.go#L1740